packages/components/src/components/lifecycleValidation.spec.ts

```diff
@@ -136,6 +148,34 @@ it.each<TagName>(tagNamesPublicWithoutProps)('should not call validateProps() fo
   expect(spy).not.toBeCalled();
 });
 
+describe.each<TagName>(tagNamesWithPropsOfTypeObject)('%s', (tagName) => {
+  const component = componentFactory(tagName);
+
+  it('should call hasPropValueChanged() with correct parameters via componentShouldUpdate', () => {
+    const spy = jest.spyOn(hasPropValueChangedUtils, 'hasPropValueChanged');
+
+    component.componentShouldUpdate('newValue', 'oldValue');
```

Comment:
Perhaps add 3rd parameter since this is how the function is always called at actual runtime.
```suggestion
    component.componentShouldUpdate('newValue', 'oldValue', 'propOrStateName');
```